packages/crypto/lib/models/block.js

```diff
@@ -321,6 +326,12 @@ module.exports = class Block {
             result.errors.push('Encountered duplicate transaction: ' + transaction.data.id)
           }
 
+          if (!transaction.verified) {
+            result.errors.push('Invalid transaction in block: ' + transaction.data.id)
+          } else if (transactionValidator.validate(transaction.data).fails) {
```

Comment:
Id keep just his part... the rest is in blockchain.js

Reply:
👌 i wanted to check for verified first to reduce any unnecessary processing with the validator, but just had the thought while testing that 0 fee could mean the transaction is "verified", but the schema could still be invalid. will remove that check and test

Reply:
That being said, if we do it later on it may have to do the whole revert transactions process. The benefit of doing it here is it doesn't start applying transactions, only to find that 1 of them is invalid. I'm going to split it into a separate `if` statement. It's only checking a boolean so speed impact will be next to nothing

Reply:
What part to you want to remove @kristjank? There is no point to run any `if` conditions outside of the first one if the transactions is already not verified, thus saving time.

Reply:
yes I agree, I would just loop the block transactions and verify them with validator... and return error or not...

everything else related to:
- wallet manager move it out....

Reply:
aa was looking at old branch...

Reply:
the only concern I have is performance, as:

- before `new Block` changes and adding the blockObject... the `new Block` was called before entering and doing the final apply check (after basic chained checks and other stuff). Now we call it on the queue level, which could be more expensive, as verify is called thru constructor. 

and we should all test this from 0 sync

Reply:
you say expensive, but before those changes it was creating a new block object 3 times when one was received from another peer. Surely that's more expensive when it has to do validation each time? 🤔 

we'll have to wait on syncing from 0, because devnet v2 will be reset again